js/feed.js

```diff
@@ -0,0 +1,301 @@
+import paramenterManagement from "../utils/utils.mjs";
+import postManagement from "../modules/posts.mjs";
+
+const API_BASE_URL = "https://api.noroff.dev/api/v1";
+
+let feedContainer = document.querySelector(".feedContainer");
+let filter = "forest";
+
+/**
+ * Initialize the functions for the DOM items
+ * @function
+ */
+function initializeFunctionality() {
+  document.getElementById("logout").addEventListener("click", logout, false);
+  document
+    .getElementById("filter-btn")
+    .addEventListener("click", toggleFilter, false);
+  document
+    .getElementById("search-btn")
+    .addEventListener("click", displayFeed, false);
+  document.getElementById("post-btn").addEventListener("click", post, false);
+}
+
+/**
+ * Toggles the filter you want to add tags to your post between two values. Future add custom additoin of
+ * tags
+ * @function
+ */
+function toggleFilter() {
+  if (filter.localeCompare("forest") === 0) {
+    filter = "ocean";
+    document.getElementById("filter-btn").innerHTML = "ocean";
+  } else if (filter.localeCompare("ocean") === 0) {
+    filter = "forest";
+    document.getElementById("filter-btn").innerHTML = "forest";
+  }
+}
+
+/**
+ * Function that displays all posts. Consider moving to modules/post, but because it is more than one I kept it here.
+ * Returns the list of 100 latest posts
+ * @function
+ */
+async function getPosts() {
+  try {
+    let uri = `${API_BASE_URL}/social/posts?_author=true`;
+    if (paramenterManagement.getParameter("filter") != null) {
+      uri = uri + "&_tag=" + paramenterManagement.getParameter("filter");
+    }
+    const response = await fetch(uri, {
+      headers: {
+        "Content-type": "application/json; charset=UTF-8",
+        Authorization: "Bearer " + localStorage.token,
+      },
+      method: "GET",
+    });
+    const resultat = await response.json();
```

Comment:
You need to handle error here. See example below

```suggestion
const resultat = await response.json();
if (!resultat.ok) {
    alert("Server respond with an invalid data...")
}
return resultat;
```